### ISSUE TITLE
Pure Tracker support (Tracker not found)

### ISSUE THREAD

**Issue by danmrc** [tpapp/LogDensityProblems.jl#53]
Hello,

First of all, this package is great. 

Second, I am having some trouble using the Flux option. It gives a Tracker not found error. I am using Julia 1.3 in Windows.

That being said, I wrote a version of the `logdensity_and_gradient` that uses just the Tracker package instead of Flux. I gave it a try and it works great with the DynamicHMC package. If you want, I can open a pull request.

Thanks in advance

**Comment by tpapp** [comment]
I think that Flux should be removed now, since it uses Zygote anyway. Cf #51.

Can you please try with Zygote?

**Comment by danmrc** [comment]
It loads with Zygote, but throws an error. I used the following example, but using Zygote instead of FowardDiff: https://tamaspapp.eu/DynamicHMCExamples.jl/latest/example_linear_regression/

Here is the error:

```
TypeError: in ccall: first argument not a pointer or valid constant expression, expected Ptr, got Tuple{String,String}
tdistlogpdf at rmath.jl:73 [inlined]
logpdf at univariates.jl:543 [inlined]
_pullback(::Zygote.Context, ::typeof(logpdf), ::TDist{Float64}, ::Float64) at interface2.jl:0
LinearRegressionProblem at untitled-b648ed369443c8d40dbcfb604ea83f7c:18 [inlined]
_pullback(::Zygote.Context, ::LinearRegressionProblem{Array{Float64,1},Array{Float64,2},Float64}, ::NamedTuple{(:β, :σ),Tuple{Array{Float64,1},Float64}}) at interface2.jl:0
transform_logdensity at generic.jl:166 [inlined]
_pullback(::Zygote.Context, ::typeof(transform_logdensity), ::TransformVariables.TransformTuple{NamedTuple{(:β, :σ),Tuple{TransformVariables.ArrayTransform{TransformVariables.Identity,1},TransformVariables.ShiftedExp{true,Float64}}}}, ::LinearRegressionProblem{Array{Float64,1},Array{Float64,2},Float64}, ::Array{Float64,1}) at interface2.jl:0
logdensity at LogDensityProblems.jl:168 [inlined]
_pullback(::Zygote.Context, ::typeof(LogDensityProblems.logdensity), ::TransformedLogDensity{TransformVariables.TransformTuple{NamedTuple{(:β, :σ),Tuple{TransformVariables.ArrayTransform{TransformVariables.Identity,1},TransformVariables.ShiftedExp{true,Float64}}}},LinearRegressionProblem{Array{Float64,1},Array{Float64,2},Float64}}, ::Array{Float64,1}) at interface2.jl:0
#40 at AD_ForwardDiff.jl:21 [inlined]
_pullback(::Zygote.Context, ::LogDensityProblems.var"#40#41"{TransformedLogDensity{TransformVariables.TransformTuple{NamedTuple{(:β, :σ),Tuple{TransformVariables.ArrayTransform{TransformVariables.Identity,1},TransformVariables.ShiftedExp{true,Float64}}}},LinearRegressionProblem{Array{Float64,1},Array{Float64,2},Float64}}}, ::Array{Float64,1}) at interface2.jl:0
_pullback at interface.jl:31 [inlined]
pullback at interface.jl:37 [inlined]
logdensity_and_gradient(::LogDensityProblems.ZygoteGradientLogDensity{TransformedLogDensity{TransformVariables.TransformTuple{NamedTuple{(:β, :σ),Tuple{TransformVariables.ArrayTransform{TransformVariables.Identity,1},TransformVariables.ShiftedExp{true,Float64}}}},LinearRegressionProblem{Array{Float64,1},Array{Float64,2},Float64}}}, ::Array{Float64,1}) at AD_Zygote.jl:21
evaluate_ℓ(::LogDensityProblems.ZygoteGradientLogDensity{TransformedLogDensity{TransformVariables.TransformTuple{NamedTuple{(:β, :σ),Tuple{TransformVariables.ArrayTransform{TransformVariables.Identity,1},TransformVariables.ShiftedExp{true,Float64}}}},LinearRegressionProblem{Array{Float64,1},Array{Float64,2},Float64}}}, ::Array{Float64,1}) at hamiltonian.jl:193
#initialize_warmup_state#20(::Array{Float64,1}, ::GaussianKineticEnergy{LinearAlgebra.Diagonal{Float64,Array{Float64,1}},LinearAlgebra.Diagonal{Float64,Array{Float64,1}}}, ::Nothing, ::typeof(DynamicHMC.initialize_warmup_state), ::Random._GLOBAL_RNG, ::LogDensityProblems.ZygoteGradientLogDensity{TransformedLogDensity{TransformVariables.TransformTuple{NamedTuple{(:β, :σ),Tuple{TransformVariables.ArrayTransform{TransformVariables.Identity,1},TransformVariables.ShiftedExp{true,Float64}}}},LinearRegressionProblem{Array{Float64,1},Array{Float64,2},Float64}}}) at mcmc.jl:109
initialize_warmup_state(::Random._GLOBAL_RNG, ::LogDensityProblems.ZygoteGradientLogDensity{TransformedLogDensity{TransformVariables.TransformTuple{NamedTuple{(:β, :σ),Tuple{TransformVariables.ArrayTransform{TransformVariables.Identity,1},TransformVariables.ShiftedExp{true,Float64}}}},LinearRegressionProblem{Array{Float64,1},Array{Float64,2},Float64}}}) at mcmc.jl:109
#mcmc_keep_warmup#33(::Tuple{}, ::Tuple{FindLocalOptimum{Float64},InitialStepsizeSearch,TuningNUTS{Nothing,DualAveraging{Float64}},TuningNUTS{LinearAlgebra.Diagonal,DualAveraging{Float64}},TuningNUTS{LinearAlgebra.Diagonal,DualAveraging{Float64}},TuningNUTS{LinearAlgebra.Diagonal,DualAveraging{Float64}},TuningNUTS{LinearAlgebra.Diagonal,DualAveraging{Float64}},TuningNUTS{LinearAlgebra.Diagonal,DualAveraging{Float64}},TuningNUTS{Nothing,DualAveraging{Float64}}}, ::DynamicHMC.NUTS{Val{:generalized}}, ::LogProgressReport{Nothing}, ::typeof(DynamicHMC.mcmc_keep_warmup), ::Random._GLOBAL_RNG, ::LogDensityProblems.ZygoteGradientLogDensity{TransformedLogDensity{TransformVariables.TransformTuple{NamedTuple{(:β, :σ),Tuple{TransformVariables.ArrayTransform{TransformVariables.Identity,1},TransformVariables.ShiftedExp{true,Float64}}}},LinearRegressionProblem{Array{Float64,1},Array{Float64,2},Float64}}}, ::Int64) at mcmc.jl:518
#mcmc_keep_warmup at none:0 [inlined]
macro expansion at Parameters.jl:750 [inlined]
#mcmc_with_warmup#34(::Tuple{}, ::Tuple{FindLocalOptimum{Float64},InitialStepsizeSearch,TuningNUTS{Nothing,DualAveraging{Float64}},TuningNUTS{LinearAlgebra.Diagonal,DualAveraging{Float64}},TuningNUTS{LinearAlgebra.Diagonal,DualAveraging{Float64}},TuningNUTS{LinearAlgebra.Diagonal,DualAveraging{Float64}},TuningNUTS{LinearAlgebra.Diagonal,DualAveraging{Float64}},TuningNUTS{LinearAlgebra.Diagonal,DualAveraging{Float64}},TuningNUTS{Nothing,Dua...
```

Using Tracker instead works flawless

**Comment by tpapp** [comment]
This should be fixed by #54.
